lib/purescript-cst/src/Language/PureScript/CST/Flatten.hs

```diff
@@ -5,12 +5,274 @@ import Prelude
 import Data.DList (DList)
 import Language.PureScript.CST.Types
 
+flattenModule :: Module a -> DList SourceToken
+flattenModule (Module _ a b c d e f g) =
+  pure a <>
+  flattenName b <>
+  foldMap (flattenWrapped (flattenSeparated flattenExport)) c <>
+  pure d <>
+  foldMap flattenImportDecl e <>
+  foldMap flattenDeclaration f <>
+  pure (SourceToken (TokenAnn dummyRange g []) TokEof)
+  where
+    dummyPos = SourcePos 0 0
+    dummyRange = SourceRange dummyPos dummyPos
```

Comment:
I wonder if the CST should keep the TokEof around instead of just the trailing module comments.

Reply:
Yeah I wondered about that as well...

Reply:
You can use `moduleRange` from `CST.Position` to get the last position. The TokEof should have the last position twice.

Reply:
`moduleRange` uses the endPosition of the last declaration in the module. Shouldn't the Eof range start after "its" leading comments?

Reply:
Yes, you can. You can use `commentDelta` and `applyDelta` to get that position. It's not great, and I should have just included TokEof in the module, but you can derive the original information.